src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,190 @@
+package duke;
+
+import duke.exception.*;
+
+import java.util.ArrayList;
+
+/**
+ * A class that represents a list of Tasks.
+ *
+ * It contains a constructor, a getter,
+ * a method to print a list, add a task to a list,
+ * mark a task as done and delete a task.
+ *
+ */
+public class TaskList {
+    private ArrayList<Task> list;
+
+    /**
+     * Constructor to create a new TaskList.
+     *
+     */
+    public TaskList() {
+        list = new ArrayList<>();
+    }
+
+    /**
+     * Constructor to create a new TaskList.
+     *
+     * @param list An ArrayList of Task that contains Tasks to be put inside the TaskList.
+     */
+    public TaskList(ArrayList<Task> list) {
+        this.list = list;
+    }
+
+    /**
+     * Method to print out tasks inside a TaskList.
+     *
+     * @return String representation of all the tasks inside the current TaskList.
+     */
+    public String printList() {
+        String str = "";
+        for (int i = 1; i <= list.size(); i++) {
+            str = str + i + ". " + list.get(i-1) + "\n";
+        }
+        System.out.println(str);
+        return str;
+    }
+
+    /**
+     * A method to mark a task as done.
+     *
+     * @param str User input which indicates details of the task to be marked as done.
+     * @return Task after it is marked as done.
+     * @throws OutOfBoundException Thrown when user inputs task number that is out of the given task range.
+     * @throws NumberFormatException Thrown when user gives a non-number as the task number.
+     */
+    public Task markAsDone(String str) throws OutOfBoundException, NumberFormatException {
+        String i = str.substring(str.length()-1);
+        int index = Integer.parseInt(i);
+        if (index < 0 || index > list.size()) {
+            throw new OutOfBoundException();
+        } else {
+            Task task = list.get(index - 1);
+            task.markAsDone();
+            System.out.println("Nice! I have marked this task as done!");
+            System.out.println(task);
+            return task;
+        }
```

Comment:
Could this logic block be refactored using a guard clause to handle the exception throwing in order to reduce indentation caused by the else block?